scratch/copy-mk.py

```diff
@@ -0,0 +1,155 @@
+    args = parser.parse_args(argv)
+    return args
+
+
+async def async_main(args) -> int:
+    cmc_client = await aiokatcp.Client.connect(args.cmc_host, args.cmc_port)
+
+    try:
+        _, informs = await cmc_client.request("subordinate-list")
+    except (aiokatcp.FailReply, ConnectionError) as exc:
+        print(f"Error: {exc}", file=sys.stderr)
+        return 1
+
+    if len(informs) == 0:
+        print(f"No subarrays currently running on {args.cmc_host}")
+        return 1
+
+    subordinates = {}
+    for inform in informs:
+        name = inform.arguments[0].decode()
+        control_port, monitor_port = inform.arguments[1].decode().split(",")
+        control_port = int(control_port)
+        monitor_port = int(monitor_port)
+        mcast_groups = []
+        for grp in inform.arguments[2:]:
+            mcast_groups.append(grp.decode())
+        n_antennas = len(mcast_groups) // 2
+        for n, grp in enumerate(mcast_groups[1:]):
+            if grp == mcast_groups[0]:
+                # MK is not using a full power-of-2 size array, there are dummies from here onwards.
+                print("Dummy antenna found starting at ", (n + 1) // 2)
+                n_antennas = (n + 1) // 2
+                break
+        digitiser_address = mcast_groups[0].split("+")[0]
+
+        subordinates[name] = Subordinate(name, control_port, monitor_port, digitiser_address, n_antennas)
+
+    try:
+        target = subordinates[args.subordinate]
+    except KeyError:
+        print(f"{args.subordinate} is not among the subordinates currently running. Options are: {subordinates.keys()}")
+        return 1
+
+    # connect to the corr2_servlet itself to see the sync time, n_chans and bandwidth
+    corr2_client = await aiokatcp.Client.connect(args.cmc_host, target.control_port)
+    sync_time = await corr2_client.sensor_value("sync-time", float)
+    channels = await corr2_client.sensor_value("antenna-channelised-voltage-n-chans", int)
+    sample_rate = await corr2_client.sensor_value("antenna-channelised-voltage-bandwidth", float) * 2
+
+    # This won't distinguish between the different kinds of S-band. It'll be wrong. I'm not quite
+    # sure at this point how to match up the centre_frequency values in katgpucbf.meerkat.BANDS with
+    # what MK actually reports, even in the L and UHF cases, so for the purposes of getting
+    # something producing a result, I am leaving it as-is for the time being.
+    band = next((k for k, v in BANDS.items() if v.adc_sample_rate == sample_rate), None)
+    if not band:
+        print(
+            f"Unable to determine which band. Sample rate reported as {sample_rate}",
+            file=sys.stderr,
+        )
+        return 1
+
+    out_kwargs = {
+        "name": target.name.replace(".", "_"),
+        "antennas": target.n_antennas,
+        "channels": channels,
+        "digitiser-address": mcast_groups[0].split("+")[0],
+        "sync-time": sync_time,
+        "band": band,
```

Comment:
I take it --beams is being left for later?

Reply:
I did sort of think of that but I couldn't see a straightforward way to figure out how many beams the MK correlator is making.

I guess it doesn't really matter though? But yeah for the purposes of this exercise I didn't think beams were needed yet. It was mainly at this stage to check that the routing works, and later on it's going to be used to check SDP while CAM may not yet be ready.